File: frontend/src/types/notification.ts
export type NotificationType = "weather" | "disease" | "crop" | "system" | "general";
export type NotificationSeverity = "info" | "low" | "moderate" | "high" | "critical";

export interface Notification {
  id: string;
  type: NotificationType;
  severity: NotificationSeverity;
  title: string;
  message: string;
  related_entity_id: string | null;
  related_entity_type: string | null;
  is_read: boolean;
  created_at: string;
}

export interface NotificationListResponse {
  notifications: Notification[];
  unread_count: number;
}

File: backend/app/enums/notification.py
# backend/app/enums/notification.py
"""
Notification Type & Severity Enums

Module: Phase 1 → Module 9 → Notifications
Author: VerdiGO Backend Team
"""

from enum import Enum


class NotificationTypeEnum(str, Enum):
    WEATHER = "weather"
    DISEASE = "disease"
    CROP = "crop"
    SYSTEM = "system"
    GENERAL = "general"


class NotificationSeverityEnum(str, Enum):
    INFO = "info"
    LOW = "low"
    MODERATE = "moderate"
    HIGH = "high"
    CRITICAL = "critical"